### PR TITLE
docs(react-router-v4): update API

### PR DESCRIPTION
**Pull request by mthuret**
See #2020

### CODE REVIEW

File: packages/react-instantsearch/examples/react-router-v4/src/App.js [hunk 1]
```diff
@@ -7,35 +7,39 @@ import {
...
     return (
       <InstantSearch
         appId="latency"
         apiKey="6be0576ff61c053d5f9a3225e2a90f76"
         indexName="ikea"
-        searchState={qs.parse(location.query)}
+        searchState={this.state.searchState}
         onSearchStateChange={this.onSearchStateChange.bind(this)}
-        createURL={state => `?${qs.stringify(state)}`}
+        createURL={this.createURL.bind(this)}
```
Comment (Haroenv) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: Why is the binding needed for an arrow function?

**Comment by mthuret** [comment]
@Haroenv: good catch for the bind. Also we don't have a preview of the recipes yet, so you will not be able to see it live through netlify :(

File: packages/react-instantsearch/examples/react-router-v4/src/App.js [hunk 2]
```diff
@@ -7,33 +7,35 @@ import {
...
+const updateAfter = 700;
+const searchStateToUrl =
+  (props, searchState) =>
+    searchState ? `${props.location.pathname}?${qs.stringify(searchState)}` : '';
+
+class App extends Component {
+  constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {searchState: {...qs.parse(props.location.search.slice(1))}};
```
Comment (Kikobeats) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: why not

```js
this.state = {searchState: qs.parse(props.location.search.slice(1))};
```

you are destructuring and create and object 😄

File: packages/react-instantsearch/examples/react-router-v4/src/App.js [hunk 3]
```diff
@@ -7,33 +7,35 @@ import {
 import 'react-instantsearch-theme-algolia/style.css';
 import qs from 'qs';
 
-export default class App extends Component {
-  constructor (props) {
+const updateAfter = 700;
+const searchStateToUrl =
+  (props, searchState) =>
+    searchState ? `${props.location.pathname}?${qs.stringify(searchState)}` : '';
```
Comment (Kikobeats) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: I extracted this method:

```js
const createURL = (state) => `?${qs.stringify(state)}`
```

for be used lately and avoid repeat this pice of code 👍

**Comment by mthuret** [comment]
Should be ok now :)

**Comment by Kikobeats** [comment]
awesome!

**Comment by Kikobeats** [comment]
I'm seeing something bad:

```
Warning: Failed prop type: The prop `push` is marked as required in `Unknown`, but its value is `undefined`.
```

and makes sense, `.push` are not in `props`: https://reacttraining.com/react-router/#route-props

I didn't detect it in the past because browser cache 😭

**Comment by Kikobeats** [comment]
`this.props.push` → `this.props.history.push`

**Comment by Haroenv** [comment]
You can do a pull request @Kikobeats <3

**Pull request merged**